libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/utils/RouteUtilsTest.java

```diff
@@ -57,65 +63,60 @@ public void isNewRoute_returnsTrueWhenGeometriesDoNotEqual() throws Exception {
   }
 
   @Test
-  public void isArrivalEvent_returnsTrueWhenManeuverTypeIsArrival_andIsValidMetersRemaining() throws Exception {
-    DirectionsRoute aRoute = buildTestDirectionsRoute();
-    int lastStepIndex = obtainLastStepIndex(aRoute);
-    RouteProgress defaultRouteProgress = buildDefaultTestRouteProgress();
-    RouteProgress theRouteProgress = defaultRouteProgress.toBuilder()
-      .stepDistanceRemaining(30)
-      .legDistanceRemaining(30)
-      .distanceRemaining(30)
-      .stepIndex(lastStepIndex)
-      .build();
-
-    boolean isArrivalEvent = RouteUtils.isArrivalEvent(theRouteProgress);
+  public void isArrivalEvent_returnsTrueWhenManeuverTypeIsArrival_andIsLastInstruction() throws Exception {
+    DirectionsRoute route = buildTestDirectionsRoute();
```

Comment:
shouldn't these all be mocked instead of test objects? Because really, this is reliant on the other implementations to work (DirectionsRoute, RouteLeg, etc.)